pkg/controller/kogitodataindex/resource/manage_resources.go

```diff
@@ -94,6 +98,99 @@ func ensureReplicas(instance *v1alpha1.KogitoDataIndex, statefulset *appsv1.Stat
 	return false
 }
 
+func ensureHTTPPort(instance *v1alpha1.KogitoDataIndex, statefulset *appsv1.StatefulSet, client *client.Client) bool {
```

Comment:
We should wait for https://github.com/kiegroup/kogito-cloud-operator/pull/133 instead. The way we are handling with updates get much simpler with `operator-utils`. @xiezhang7 your PR is handling statefulsets differences, right?

Reply:
The deployed and requested StatefulSets, Services and Routes are compared using `operator-utils` respectively. `manage_resources.go` is no longer needed.